chore/release/release.go

```diff
@@ -0,0 +1,150 @@
+package chore
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/cli/go-gh/v2/pkg/api"
+	copyright "github.com/coreruleset/crs-toolchain/v2/chore/update_copyright"
+	"github.com/coreruleset/crs-toolchain/v2/context"
+	"github.com/go-git/go-git/v5"
+	"github.com/rs/zerolog/log"
+)
+
+var logger = log.With().Str("component", "release").Logger()
+
+func Release(context *context.Context, repositoryPath string, version *semver.Version, sourceRef string) {
+	remoteName := findRemoteName()
+	if remoteName == "" {
+		logger.Fatal().Msg("failed to find remote for coreruleset/coreruleset")
+	}
+	fetchSourceRef(remoteName, sourceRef)
+	branchName := fmt.Sprintf("v%d.%d.%d", version.Major(), version.Minor(), version.Patch())
+	createAndCheckOutBranch(context, branchName, sourceRef)
+	copyright.UpdateCopyright(context, version, uint16(time.Now().Year()), []string{"util/crs-rules-check/examples"})
+	createCommit(context, branchName)
+	pushBranch(remoteName, branchName)
+	createPullRequest(version, branchName, sourceRef)
+}
+
+func createAndCheckOutBranch(context *context.Context, branchName string, sourceRef string) {
+	if err := checkForCleanWorkTree(context); err != nil {
+		// FIXME
+		panic(err)
+	}
+
+	out, err := runGit(context.RootDir(), "switch", "-c", branchName, sourceRef)
+	if err != nil {
+		logger.Fatal().Err(err).Bytes("command-output", out).Msg("failed to create commit for release")
+	}
+}
+
+func createCommit(context *context.Context, branchName string) {
+	out, err := runGit(context.RootDir(), "commit", "-am", "chore: release "+branchName)
+	if err != nil {
+		logger.Fatal().Err(err).Bytes("command-output", out).Msg("failed to create commit for release")
+	}
+}
+
+func checkForCleanWorkTree(context *context.Context) error {
+	repositoryPath := context.RootDir()
+	repo, err := git.PlainOpen(repositoryPath)
+	if err != nil {
+		//FIXME
+		panic(err)
+	}
+	worktree, err := repo.Worktree()
+	if err != nil {
+		//FIXME
+		panic(err)
+	}
+	status, err := worktree.Status()
+	if err != nil {
+		//FIXME
+		panic(err)
+	}
+	if !status.IsClean() {
+		// FIXME
+		return errors.New("worktree not clean. Please stash or commit your changes first")
+	}
+	return nil
+}
+
+func createPullRequest(version *semver.Version, branchName string, targetBranchName string) {
+	opts := api.ClientOptions{
+		Headers: map[string]string{"Accept": "application/octet-stream"},
+	}
+	client, err := api.NewRESTClient(opts)
+	if err != nil {
+		log.Fatal().Err(err).Send()
+	}
+
+	type prBody struct {
+		Title    string `json:"title"`
+		Head     string `json:"head"`
+		Base     string `json:"base"`
+		Label    string `json:"label"`
+		Reviewer string `json:"reviewer"`
+	}
+	bodyJson, err := json.Marshal(&prBody{
+		Title:    fmt.Sprintf("Release v%d.%d%d", version.Major(), version.Minor(), version.Patch()),
```

Comment:
```suggestion
		Title:    fmt.Sprintf("chore: release v%d.%d%d", version.Major(), version.Minor(), version.Patch()),
```
otherwise it will fail at the pr title check